pkg/server/plugin/upstreamauthority/awspca/pca_client_fake.go

```diff
@@ -23,36 +23,46 @@ type pcaClientFake struct {
 	expectedGetCertificateInput *acmpca.GetCertificateInput
 	getCertificateOutput        *acmpca.GetCertificateOutput
 	getCertificateErr           error
-
-	waitUntilCertificateIssuedErr error
 }
 
-func (f *pcaClientFake) DescribeCertificateAuthorityWithContext(ctx aws.Context, input *acmpca.DescribeCertificateAuthorityInput, option ...request.Option) (*acmpca.DescribeCertificateAuthorityOutput, error) {
+func (f *pcaClientFake) DescribeCertificateAuthority(ctx context.Context, input *acmpca.DescribeCertificateAuthorityInput, optFns ...func(*acmpca.Options)) (*acmpca.DescribeCertificateAuthorityOutput, error) {
 	require.Equal(f.t, f.expectedDescribeInput, input)
 	if f.describeCertificateErr != nil {
 		return nil, f.describeCertificateErr
 	}
 	return f.describeCertificateOutput, nil
 }
 
-func (f *pcaClientFake) IssueCertificateWithContext(ctx aws.Context, input *acmpca.IssueCertificateInput, option ...request.Option) (*acmpca.IssueCertificateOutput, error) {
+func (f *pcaClientFake) IssueCertificate(ctx context.Context, input *acmpca.IssueCertificateInput, optFns ...func(*acmpca.Options)) (*acmpca.IssueCertificateOutput, error) {
 	require.Equal(f.t, f.expectedIssueInput, input)
 	if f.issueCertifcateErr != nil {
 		return nil, f.issueCertifcateErr
 	}
 	return f.issueCertificateOutput, nil
 }
 
-func (f *pcaClientFake) WaitUntilCertificateIssuedWithContext(ctx aws.Context, input *acmpca.GetCertificateInput, option ...request.WaiterOption) error {
+func (f *pcaClientFake) GetCertificate(ctx context.Context, input *acmpca.GetCertificateInput, optFns ...func(*acmpca.Options)) (*acmpca.GetCertificateOutput, error) {
+	require.Equal(f.t, f.expectedGetCertificateInput, input)
+	if f.getCertificateErr != nil {
+		return nil, f.getCertificateErr
+	}
+	return f.getCertificateOutput, nil
+}
+
+type fakeCertificateIssuedWaiter struct {
```

Comment:
This isn't too big of a deal, and I haven't dug deep here, but I was curious the decision to hook the creation of the waiter to hook the behavior v.s. just using the fake client GetCertificate. The real "waiter" just calls GetCertificate in a loop. ~Seems like we could just use the return values from GetCertificate to influence the code under test.~

Scratch that. I went down the rabbit hole and discovered that the certificate waiter always retries, independent of the error returned from GetCertificate, which would make it unwieldy to get the `Wait` call to return without some heroics using the default behavior. I've filed https://github.com/aws/aws-sdk-go-v2/issues/1585.

We could probably still get mostly there by setting the Retryable option on the waiter to implement what I _think_ is the correct behavior (see the issue above).